### ISSUE TITLE
Support more heading levels

### ISSUE THREAD

**Issue by 42sol-eu** [MartinPacker/mdpre#4]
in `def extractTableOfContents(input_file):` the heading level could be calculated to improve code readability:

```python
for line in input_file:
        if line.startswith("#"):
            title = line.lstrip('#')
            level = len(line) - len(title)
            title = title.lstrip().rstrip()
            print(chars,title)
            # followed by min and max checking 
```

This has the second advantage that the character for heading could be easily put as a functions parameter (planing for extending support to dialects)

**Comment by MartinPacker** [comment]
Do you want to do this one? If so please create a branch to work on.

**Comment by 42sol-eu** [comment]
if you like the idea - sure

**Comment by MartinPacker** [comment]
Yes please. I think one of the benefits you're suggesting is no limit on the number of levels of heading. While I might never go beyond 5, I think removing an arbitrary limit is goodness.

I'm doing nothing on mdpre right now so please consider it yours to work on.

**Comment by MartinPacker** [comment]
Thanks for the code. Merged and tested and it's the start of v0.4.6.
